### PR TITLE
[WIP] Make cli portable

### PR DESCRIPTION
**Pull request by pakrym**
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2036)

### CODE REVIEW

**Comment by piotrpMSFT** [comment]
@dotnet-bot test OSX x64 Release Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test OSX x64 Release Build please.

**Comment by piotrpMSFT** [comment]
@mmitche looks like our OS X build agent is not feeling well this fine day. Anything we can do to get it back on its feet?

**Comment by mmitche** [comment]
@piotrpMSFT We might try rebooting them.  The logs seem to indicate that it's trying to connect twice for some reason.

**Comment by piotrpMSFT** [comment]
@mmitche do we have instructions for rebooting them?

**Comment by mmitche** [comment]
Try to just retrigger and let me know if it happens again.  The reboot didn't help on other machines so I restarted the whole system.

Sent from my Windows Phone
________________________________

**Comment by piotrpMSFT** [comment]
@dotnet-bot test OSX x64 Release Build please.

**Comment by piotrpMSFT** [comment]
Trying again. Looks like the build agent stopped sending back data...

**Comment by piotrpMSFT** [comment]
@dotnet-bot test OSX x64 Release Build please.

**Comment by mmitche** [comment]
I'm not sure about that.  I saw a bunch of other processes stop around that time while talking to myget.  Others were humming along.

Sent from my Windows Phone
________________________________

File: scripts/dotnet-cli-build/Utils/Utils.cs [hunk 1]
```diff
@@ -96,7 +96,7 @@ public static void DeleteDirectory(string path)
                     File.SetAttributes(file, FileAttributes.Normal);
                     File.Delete(file);
                 }
-
+                System.Threading.Thread.Sleep(1);
```
Comment (pakrym) [inline scripts/dotnet-cli-build/Utils/Utils.cs]: Solved the issue when all files inside directory were deleted but directory deletion failed with "non empty" error.

**Comment by pakrym** [comment]
@dotnet-bot test OSX x64 Release Build please
@dotnet-bot test Windows_NT x64 Release Build please
@dotnet-bot test Windows_NT x86 Debug Build please

**Comment by anurse** [comment]
@dotnet-bot test OSX x64 Release Build

**Comment by pakrym** [comment]
@dotnet-bot test this please

**Comment by pakrym** [comment]
@dotnet-bot test this please

File: src/compilers/project.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        "version":"1.0.0-rc2-23925"
+      },
+      "Microsoft.CodeAnalysis.CSharp": "1.3.0-beta1-20160321-04",
+      "Microsoft.Net.Compilers.netcore": "1.3.0-beta1-20160321-04",
+      "Microsoft.Net.CSharp.Interactive.netcore": "1.3.0-beta1-20160321-04",
+      "Microsoft.FSharp.Compiler.netcore": "1.0.0-alpha-160318",
+      "Microsoft.DiaSymReader.Native": "1.3.3",
+    },
+    "frameworks": {
+      "netstandard1.5": {
```
Comment (weshaggard) [inline src/compilers/project.json]: Shouldn't this be `netcoreapp1.0`? @anurse now that we support `type:platform` what is there anything preventing us from using `netcoreapp1.0` now?

Comment (anurse) [inline src/compilers/project.json]: RID inference is still on. I'm working on removing that.

**Comment by pakrym** [comment]
@dotnet-bot test this please

**Comment by pakrym** [comment]
@dotnet-bot test RHEL7.2 x64 Release Build please

**Comment by anurse** [comment]
:shipit:

File: src/dotnet/project.json [hunk 3]
```diff
@@ -64,10 +62,10 @@
     }
   },
   "frameworks": {
-    "netstandardapp1.5": {
+    "netstandard1.5": {
       "imports": [
         "dnxcore50",
-        "portable-net45+win8"
```
Comment (livarcocc) [inline src/dotnet/project.json]: We need this import for DiaSymReader that we use in TestingAbstractions to read full pbds. Why did you remove it here?

File: test/ArgumentsReflector/project.json [hunk 4]
```diff
@@ -5,6 +5,7 @@
   },
   "dependencies": {
     "NETStandard.Library": "1.5.0-rc2-23928"
```
Comment (anurse) [inline test/ArgumentsReflector/project.json]: Missing comma :). That's also what's causing the build failure.

Comment (pakrym) [inline test/ArgumentsReflector/project.json]: What took you so long? :)

**Comment by piotrpMSFT** [comment]
Wohoo!

/cc @petermarcu

**Comment by Petermarcu** [comment]
Nice work! Great to see this go in! I'll be waiting for the builds to download and play with it! :)
